base/plugins/score-plugin-engine/Engine/Protocols/Joystick/JoystickSpecificSettingsSerialization.cpp

```diff
@@ -0,0 +1,34 @@
+
+#include "JoystickSpecificSettings.hpp"
+
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QString>
+#include <score/serialization/DataStreamVisitor.hpp>
+#include <score/serialization/JSONVisitor.hpp>
+
+
+
+template <>
+void DataStreamReader::read(const Engine::Network::JoystickSpecificSettings& n)
+{
```

Comment:
shouldn't at least the id be serialized ? what happens today when saving / reloading ?

Reply:
The Id is an Instance Id which is only guaranteed to be unique trough one execution. SDL provide also joystick GUID we could had this to settings and use that to serialize.

Reply:
*add